docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+        .build();
+
+// creating Meter with InstrumentationScope, comprising of
+// name, schema and attributes.
+let meter = global::meter_with_scope(scope);
+
+// creating Meter with just name
+let meter = global::meter("my_company.my_product.my_library");
+```
+
+### Instruments
+
+:heavy_check_mark: You should understand and pick the right instrument type.
+
+  > [!NOTE] Picking the right instrument type for your use case is crucial to
+  > ensure the correct semantics and performance. Check the [Instrument
+    Selection](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection)
+    section from the supplementary guidelines for more information.
+
+| OpenTelemetry Specification | OTel Rust Instrument Type |
+| --------------------------- | -------------------- |
+| [Asynchronous Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter) | [`ObservableCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableCounter.html) |
+| [Asynchronous Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-gauge) | [`ObservableGauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableGauge.html) |
+| [Asynchronous UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-updowncounter) | [`ObservableUpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableUpDownCounter.html) |
+| [Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#counter) | [`Counter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Counter.html) |
+| [Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#gauge) | [`Gauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Gauge.html) |
+| [Histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#histogram) | [`Histogram`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Histogram.html) |
+| [UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#updowncounter) | [`UpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.UpDownCounter.html) |
+
+:stop_sign: You should avoid creating instruments (e.g. `Counter`) too
+frequently. Instruments are fairly expensive and meant to be reused. For most
+applications, instruments can be created once and re-used. Instruments can also
+be cloned to create multiple handles to the same instrument, but the cloning
+should not be on hot path, but instead the cloned instance should be stored and
```

Comment:
It's not that bad to clone the instrument as it should only be a matter of incrementing the Arc atomic value, right?

Reply:
its not as bad as creating new one repeatedly, but the best option is to create/clone and re-use.

Reply:
I agree that the best thing to do is create/clone and re-use. In my opinion, I find exaggerating the dangers of a not-so-harmful operation a bit odd. We ideally want to mention only those things that we want the users to look out for and not overstate things to lessen the burden on the user.